kvdb-rocksdb/examples/memtest.rs

```diff
@@ -0,0 +1,111 @@
+use kvdb_rocksdb::{DatabaseConfig, Database};
+use ethereum_types::H256;
+use keccak_hash::keccak;
+use sysinfo::{get_current_pid, ProcessExt, System, SystemExt};
+
+const COLUMN_COUNT: u32 = 100;
+
+#[derive(Clone)]
+struct KeyValueSeed {
+    seed: H256,
+    key: H256,
+    val: H256,
+}
+
+fn next(seed: H256) -> H256 {
+    let mut buf = [0u8; 33];
+    buf[0..32].copy_from_slice(&seed[..]);
+    buf[32] = 1;
+
+    keccak(&buf[..])
+}
+
+impl KeyValueSeed {
+    fn with_seed(seed: H256) -> Self {
+        KeyValueSeed {
+            seed,
+            key: next(seed),
+            val: next(next(seed)),
+        }
+    }
+
+    fn new() -> Self {
+        Self::with_seed(H256::random())
+    }
+}
+
+impl Iterator for KeyValueSeed {
+    type Item = (H256, H256);
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let result = (self.key, self.val);
+        self.key = next(self.val);
+        self.val = next(self.key);
+
+        Some(result)
+    }
+}
+
+fn proc_memory_usage() -> u64 {
+    let mut sys = System::new();
+    let self_pid = get_current_pid().ok();
+    let memory = if let Some(self_pid) = self_pid {
+        if sys.refresh_process(self_pid) {
+            let proc = sys.get_process(self_pid)
+                .expect("Above refresh_process succeeds, this should be Some(), qed");
+            proc.memory()
+        } else { 0 }
+    } else { 0 };
+
+    memory
+}
+
+fn main() {
+
+    let mb_per_col = std::env::args()
+        .nth(1)
+        .map(|arg| arg.parse().expect("Megabytes per col - should be integer or missing"))
+        .unwrap_or(1);
+
+
+    let mut config = DatabaseConfig::with_columns(COLUMN_COUNT);
+
+    for c in 0..=COLUMN_COUNT {
+        config.memory_budget.insert(c, mb_per_col);
+    }
+    let dir = tempdir::TempDir::new("rocksdb-example").unwrap();
+
+    println!("Database is put in: {} (maybe check if it was deleted)", dir.path().to_string_lossy());
```

Comment:
I think tempdirs are deleted as soon as the process ends, so not sure what you mean by "maybe check if it was deleted"?

Reply:
Well it is not deleted when I ctrl-c currently 🤷‍♀️